rules/network/execution_potential_rce_via_toolshell.toml

```diff
@@ -0,0 +1,70 @@
+[metadata]
+creation_date = "2025/07/23"
+integration = ["network_traffic"]
+maturity = "production"
+updated_date = "2025/07/23"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects potential remote code execution (RCE) attempts targeting IIS web servers running SharePoint via malicious
+VIEWSTATE payloads in HTTP POST requests. Attackers may exploit insecure deserialization in the VIEWSTATE parameter to
+execute arbitrary code. This rule identifies suspicious requests containing VIEWSTATE data and other indicators of
+exploitation, specifically those associated with the Toolshell exploit chain. Toolshell leverages vulnerabilities
+(CVE-2025-53770 and CVE-2025-53771) for initial access, enabling adversaries to deploy a webshell, steal machine keys,
+sign VIEWSTATE payloads offline, and subsequently send signed payloads to the server to achieve code execution.
+"""
+from = "now-9m"
+index = ["logs-network_traffic.http*"]
+language = "kuery"
+license = "Elastic License v2"
+max_signals = 10
+name = "Potential VIEWSTATE RCE Attempt on SharePoint/IIS"
+references = [
+    "https://research.eye.security/sharepoint-under-siege/",
+    "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2025-53771",
+    "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2025-53770",
+    "https://msrc.microsoft.com/blog/2025/07/customer-guidance-for-sharepoint-vulnerability-cve-2025-53770/",
+]
+risk_score = 73
+rule_id = "99c9af5a-67cf-11f0-b69e-f661ea17fbcd"
+setup = """### Network Traffic Setup
+
```

Comment:
once you add an investigation guide, it would be nice to cover the FPs.

Reply:
This one may have some FPs, not sure as we don't have a production IIS environment to run the query against to establish baseline but I suspect they will be low. The other rule, FPs should be low. It's more noise I'd be concerned about which is why I added a max number of signals.